### ISSUE TITLE
Failed deleting solution Center of Excellence - Core Components (Upgrade). Solution dependencies exist, cannot uninstall.

### ISSUE THREAD

**Issue by RickKush** [microsoft/coe-starter-kit#7376]
Trying to delete CenterofExcellenceCoreComponents_Upgrade. It shows this dependency: Microsoft Dataverse (legacy) CenterofExcellenceCoreComponents-a3c3f. I deleted both of the Dataverse connections...

AB#1941

**Comment by RickKush** [comment]
I have gotten past the DV connection dependency. Now I'm stuck on an environment variable dependency, Get Security Roles for All BUs, which is used in the Admin | Sync Template v3 (Environment Properties) Flow.

**Comment by RickKush** [comment]
I reimported CenterofExcellenceCoreComponents 4.8 (the previous version) successfully, and I still cant delete 4.17 Upgrade.

**Comment by Jenefer-Monroe** [comment]
Hello. I'm not sure what state you re in now. 

Can you please share a screenshot like this so we can see what solutions you have installed in this environment? Be sure to show all records (all rows) and expand the indicated columns so that we can see full names
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/8dd512cc-21fc-4c64-adab-0f64c29b9bf2)

If you go to history you can find where the failure occurred and download details about it.
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/eb0966a6-dbfb-400b-b796-5ecb5d20c1da)

**Comment by RickKush** [comment]
![image](https://github.com/microsoft/coe-starter-kit/assets/116654106/d61d34f3-29d2-4468-a1ef-ecfeb55eaf03)

**Comment by RickKush** [comment]
![image](https://github.com/microsoft/coe-starter-kit/assets/116654106/c771a748-34fa-47d0-bdff-90311fe81d26)

**Comment by RickKush** [comment]
Solution dependencies exist, cannot uninstall. DependencyCount : 1 RequiredComponentObject details: Type: EnvironmentVariableDefinition, ObjectName: admin_GetSecurityRolesforAllBUs, Id: 51198898-00c9-499d-8a0d-583a260f710e, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: Workflow, ObjectName: Admin | Sync Template v3 (Environment Properties), Id: 546a16ff-3f13-ed11-b83d-00224823a71f, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000

**Comment by Jenefer-Monroe** [comment]
OK, you are upgrading from May 2023 to Sept 2023. Note that its difficult for us to assist with upgrade issues from this far back as it really requires knowledge of the state of the builds and my memory isnt that good. But hopefully we can get you unblocked here.

You will for sure hit an issue with the Audit Logs solution here as it has been deprecated. Please see https://github.com/microsoft/coe-starter-kit/issues/6018
Unfortunately step one of that is deleting that _upgrade which you are unable to do.

One option would be for you to just delete the contents of the problematic flows which would create an unmanaged layer but allow the delete of _upgrade. Then we can clean up the layers after everything is resolved. 

So please open Admin | Sync Template v3 (Environment Properties), delete the scope shown, save, and try to delete the _Upgrade again
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/bb1f5781-e2dc-416e-adf4-b09be6ff5980)

**Comment by RickKush** [comment]
I was able to run the delete, and it timed out after 5 minutes: "No response received: timeout of 300000ms exceeded (ECONNABORTED)." But in the history, it doesn't show failed and it won't let me try again because it thinks that this is still currently running: "Cannot start another [Uninstall] because there is a previous [Uninstall] running at this moment. Use Solution History for more details. -- The solution installation or removal failed due to the installation or removal of another solution at the same time. Please try again later."

![image](https://github.com/microsoft/coe-starter-kit/assets/116654106/e4b46280-1503-4605-b7ac-83b4661ecf18)

**Comment by Jenefer-Monroe** [comment]
Yes unfortunately the UX times out and gives that error. Its a bummer of a bug in the product's ALM. 
You have to wait for the uninstall to complete, it shoudl give a success or failure in the history once its finally done.

**Comment by RickKush** [comment]
How long to wait? It already says ECONNABORTED.

**Comment by Jenefer-Monroe** [comment]
It really depends on the build and the service at the time. Sometimes uninstall can take 30-45 minutes

**Comment by RickKush** [comment]
I'll give it an hour and let you know.

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team
